src/app/components/file-input/file-input.component.ts

```diff
@@ -1,68 +1,92 @@
-  public fileInput: ElementRef<HTMLInputElement>;
-  @Output()
-  public events = new EventEmitter<IFileSelectedEvent>();
-  private _selectedFile: File;
-
-  constructor() { }
-
-  public fileSelected(input: any): void {
-    if (input && input.target && input.target.files) {
-      const inputFile = input.target.files[0];
-      console.log(`Filename: ${inputFile.name}, Size: ${inputFile.size} bytes`);
-      if (inputFile.size > 10 * 1024 * 1024) {
-        console.log(`The selected file is too large`);
-        this.fileInput.nativeElement.value = '';
-        return;
-      }
-      this._selectedFile = inputFile;
-    }
-  }
-
-  public readFile() {
-    if (this.selectedFile) {
-      /**
-       * TODO: save the loaded text in memory or read in chunks for them to be later emitted with a the DataStreamService.
-       */
-      const reader = new FileReader();
-      reader.onload = (event) => { this.fileLoaded(event); };
-      reader.onprogress = this.logProgress;
-      reader.readAsText(this._selectedFile);
-    } else {
-      console.log(`No file selected!`);
-    }
-  }
-
-  private fileLoaded(ev: ProgressEvent): void {
-    const loadedFile = (ev.target as FileReader).result;
-    const loadedText = loadedFile.toString().split(/\r\n|\n/);
-    console.log(`File loaded! It has ${loadedText.length} lines`);
-    this.events.emit({
-      filename: this.selectedFile.name,
-    });
-  }
-
-  private logProgress(ev: ProgressEvent): void {
-    console.log(`Loading...`);
-  }
-
-  get selectedFile(): File | undefined {
-    return this._selectedFile;
-  }
-
-}
+import { Component, ViewChild, ElementRef, Output, EventEmitter } from '@angular/core';
+import { IFileSelectedEvent } from 'src/app/common';
+import { DataStreamService } from 'src/app/services';
+
+/**
+ * Component for loading the selected file.
+ *
+ * @see https://developer.mozilla.org/en-US/docs/Web/API/FileReader
+ */
+@Component({
+  selector: 'app-file-input',
+  templateUrl: './file-input.component.html',
+  styleUrls: ['./file-input.component.sass']
+})
+export class FileInputComponent {
+  @ViewChild('fileInput')
+  public fileInput: ElementRef<HTMLInputElement>;
+
+  @ViewChild('fileInputSpan')
+  public fileInputSpan: ElementRef<HTMLInputElement>;
+
+  @Output()
+  public events = new EventEmitter<IFileSelectedEvent>();
+  private _selectedFile: File;
+  private _dataSignal: string[];
+
+  constructor(private dataStreamService: DataStreamService) {}
+
+  public fileSelected(input: any): void {
+    const csvRegexp = /(.+\.csv|.+\.CSV)/;
+
+    if (input && input.target && input.target.files) {
+      const inputFile = input.target.files[0];
+      console.log(`Filename: ${inputFile.name}, Size: ${inputFile.size} bytes`);
+      if (inputFile.size > 10 * 1024 * 1024) {
+        console.log(`The selected file is too large`);
+        this.fileInput.nativeElement.value = '';
+        return;
+      }
+
+      if (csvRegexp.test(inputFile.name) === false) {

```

Comment:
```suggestion
      if (intputFile.name.endsWith('.csv')) {
```
Why not use the endsWith() function?